ui/lib/kv/addon/routes/secret/details.js

```diff
@@ -18,22 +18,14 @@ export default class KvSecretDetailsRoute extends Route {
 
   model(params) {
     const parentModel = this.modelFor('secret');
-    // Only fetch versioned data if selected version does not match parent (current) version
-    // and parentModel.secret has failReadErrorCode since permissions aren't version specific
-    if (
-      params.version &&
-      parentModel.secret.version !== params.version &&
-      !parentModel.secret.failReadErrorCode
-    ) {
-      // query params have changed by selecting a different version from the dropdown
-      // fire off new request for that version's secret data
-      const { backend, path } = parentModel;
-      return hash({
-        ...parentModel,
-        secret: this.store.queryRecord('kv/data', { backend, path, version: params.version }),
-      });
-    }
-    return parentModel;
+    const { backend, path } = parentModel;
+    // if a version is selected from the dropdown it triggers a model refresh
+    // and we fire off new request for that version's secret data
+    const query = { backend, path, ...(params.version ? { version: params.version } : null) };
```

Comment:
nit: personally I find this a bit hard to read. but could be pre-coffee brain
```suggestion
    const query = { backend, path };
    if (params.version) {
      query.version = params.version;
    }
```

Reply:
Optional spreading can be awkward to read! I'll update